anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -820,26 +820,26 @@ def _switch_active_grypedb(
         inactive_records = self._find_match(db, active=False)
         for inactive_record in inactive_records.all():
             catalog_client.delete_document(
-                inactive_record.group_name, inactive_record.checksum
+                inactive_record.group_name, inactive_record.archive_checksum
             )
         inactive_records.delete(synchronize_session="evaluate")
         # search for active and mark inactive
         self._find_match(db, active=True).update(
-            {GrypeDBMetadata.active: False}, synchronize_session="evaluate"
+            {GrypeDBFeedMetadata.active: False}, synchronize_session="evaluate"
         )
         # insert new as active
         object_url = catalog_client.create_raw_object(
             group_download_result.group, checksum, record.data
         )
-        date_generated = rfc3339str_to_datetime(record.metadata["built"])
-        grypedb_meta = GrypeDBMetadata(
-            checksum=checksum,
+        built_at = rfc3339str_to_datetime(record.metadata["built"])
+        grypedb_meta = GrypeDBFeedMetadata(
+            archive_checksum=checksum,
+            metadata_checksum=None,
```

Comment:
Maybe when we implement the change that @nightfurys was talking about last week of having the feeds update process pass the db to the wrapper to do a temp build to verify that there are no issues with it, the grype wrapper can return the metadata checksum. This would also allow us to make the metadata_checksum non nullable

Reply:
yep, that's the idea. the nullable change is temporary